File: sceneview/src/main/java/io/github/sceneview/collision/QuaternionEvaluator.java
package io.github.sceneview.collision;

import android.animation.TypeEvaluator;

/** TypeEvaluator for Quaternions. Used to animate rotations. */
public class QuaternionEvaluator implements TypeEvaluator<Quaternion> {
  @Override
  public Quaternion evaluate(float fraction, Quaternion startValue, Quaternion endValue) {
    return Quaternion.slerp(startValue, endValue, fraction);
  }
}

File: sceneview/src/main/java/io/github/sceneview/collision/QuaternionEvaluator.kt
package io.github.sceneview.collision

import android.animation.TypeEvaluator

/** TypeEvaluator for Quaternions. Used to animate rotations. */
class QuaternionEvaluator : TypeEvaluator<Quaternion> {
    override fun evaluate(fraction: Float, startValue: Quaternion, endValue: Quaternion): Quaternion {
        return Quaternion.slerp(startValue, endValue, fraction)
    }
}
